src/Piral.Blazor.Core/ComponentActivationService.cs

```diff
@@ -124,44 +120,26 @@ private Type GetComponent(string componentName)
             _services.TryGetValue(componentName, out var value);
             return value;
         }
-        
-        private IReadOnlyCollection<Type> FilterAttributeTypes(IDictionary<string, object> args)
-        {
-            var types = AttributeTypes;
-
-            types = FilterPages(args, types);
-            // potentially more filters here
-
-            return types;
-        }
 
-        private IReadOnlyCollection<Type> FilterPages(IDictionary<string, object> args, IReadOnlyCollection<Type> types)
+        private static IEnumerable<string> GetComponentNamesToRegister(Type member, IEnumerable<Type> attributeTypes)
         {
-            if (args.HasExplicitFlag(true, "includePages"))
-                return types; //don't filter anything out
-
-            _logger.LogInformation(
-                "Pages decorated with the @page directive are not included in the Blazor references. If this is unintended, reference the docs."
-            );
-
-            return types.Where(at => at != typeof(RouteAttribute)).ToList().AsReadOnly(); // filter out the pages
-        }
-        
-        private static IEnumerable<string> GetAllAttributeValues(Type member, IEnumerable<Type> attributeTypes)
-        {
-            return attributeTypes.Select(at=> GetAttributeValue(member, at)).Where(val => val != null);
+            return attributeTypes.Select(at => GetComponentNameToRegister(member, at)).Where(val => val != null);
         }
 
-        private static string GetAttributeValue(Type member, Type attributeType)
+        private static string GetComponentNameToRegister(Type member, Type attributeType)
         {
-            string value = attributeType.Name switch
+            switch (attributeType)
             {
-                nameof(RouteAttribute) => member.GetCustomAttribute<RouteAttribute>(false)?.Template,
-                nameof(ExposePiletAttribute) => member.GetCustomAttribute<ExposePiletAttribute>(false)?.Name,
-                _ => null
-            };
-
-            return value;
+                case Type _ when attributeType == typeof(RouteAttribute):
+                    string template = member.GetCustomAttribute<RouteAttribute>(false)?.Template;
+                    return template is null
+                        ? null
+                        : $"page-{Sanitize(template)}"; //pages have a "page-" prefix and get sanitized

```

Comment:
@FlorianRappl  I did not go with `page-${original-page-route}`, but sanitized the route first. This is because spiral-blazor has this line:

https://github.com/smapiot/piral/blob/89b2928224f69d0cc534f02f3a9d2d8ffb5d9245/src/converters/piral-blazor/src/converter.ts#L38

Which fails if the id given contains characters that are not allowed in css selectors (like `/` etc)

This can be resolved by using `getElementById`, or only somehow only sanitizing the HTML id's. But for now, I sanitize the component name itself. I use the same regex in the codegen too.

Would like your thoughts on this.

Reply:
That's great ahead thinking. Yeah the sanitization makes sense. We only need to be careful to apply it consistently.